tensorflow/compiler/xla/service/gpu/target_util.cc

```diff
@@ -354,6 +359,27 @@ unsigned int GetSharedMemoryAddressSpace(llvm::Module* module) {
   return 0;
 }
 
+void MarkKernelAsGPUKernel(llvm::Module* module, llvm::Function* kernel, 
```

Comment:
@sumesh13 "MarkKernelAsGPUKernel" seems like a palindrome. How about `AnnotateFunctionAsGpuKernel`? we are basically annotating an LLVM Function within a Module as a GPU kernel for NVPTX and AMDGPU. So the 2nd parameter should really be named as a `llvm::Function* func`.

Reply:
Sure..Annotate makes better sense. Kernel might be more appropriate since this is inside "BuildkernelPrototype" and the function is referred to as kernel in rest of the code.  The only thing this annotation does is mark it per PTX/AMD ..otherwise its is assumed to be a kernel already.

Reply:
got it.